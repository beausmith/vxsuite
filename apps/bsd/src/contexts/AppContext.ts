import { ElectionDefinition } from '@votingworks/types';
import { MemoryStorage, Storage, usbstick } from '@votingworks/utils';
import { createContext } from 'react';
import { MachineConfig } from '../config/types';

interface AppContextInterface {
  usbDriveStatus: usbstick.UsbDriveStatus;
  usbDriveEject: () => void;
  machineConfig: MachineConfig;
  electionDefinition?: ElectionDefinition;
  electionHash?: string;
  storage: Storage;
  lockMachine: () => void;
}

const appContext: AppContextInterface = {
  usbDriveStatus: usbstick.UsbDriveStatus.absent,
  usbDriveEject: () => undefined,
  machineConfig: { machineId: '0000', bypassAuthentication: false },
  electionDefinition: undefined,
  electionHash: undefined,
  storage: new MemoryStorage(),
  lockMachine: () => undefined,
};

const AppContext = createContext(appContext);

export default AppContext;
