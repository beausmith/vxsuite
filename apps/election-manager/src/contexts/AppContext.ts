import { createContext, RefObject } from 'react'
import {
  ElectionDefinition,
  SaveElection,
  PrintedBallot,
  ISO8601Timestamp,
  FullElectionTally,
  OptionalFullElectionExternalTally,
  OptionalFile,
} from '../config/types'
import { UsbDriveStatus } from '../lib/usbstick'
import CastVoteRecordFiles, {
  SaveCastVoteRecordFiles,
} from '../utils/CastVoteRecordFiles'
import { getEmptyFullElectionTally } from '../lib/votecounting'

export interface AppContextInterface {
  castVoteRecordFiles: CastVoteRecordFiles
  electionDefinition?: ElectionDefinition
  configuredAt: ISO8601Timestamp
  isOfficialResults: boolean
  printBallotRef?: RefObject<HTMLElement>
  saveCastVoteRecordFiles: SaveCastVoteRecordFiles
  saveElection: SaveElection
  setCastVoteRecordFiles: React.Dispatch<
    React.SetStateAction<CastVoteRecordFiles>
  >
  saveIsOfficialResults: () => void
  usbDriveStatus: UsbDriveStatus
  usbDriveEject: () => void
  addPrintedBallot: (printedBallot: PrintedBallot) => void
  printedBallots: PrintedBallot[]
  fullElectionTally: FullElectionTally
  fullElectionExternalTally: OptionalFullElectionExternalTally
  externalVoteRecordsFile: OptionalFile
  isTabulationRunning: boolean
  setFullElectionTally: React.Dispatch<React.SetStateAction<FullElectionTally>>
  setFullElectionExternalTally: React.Dispatch<
    React.SetStateAction<OptionalFullElectionExternalTally>
  >
  saveExternalVoteRecordsFile: (externalFile: OptionalFile) => void
  setIsTabulationRunning: React.Dispatch<React.SetStateAction<boolean>>
}

const appContext: AppContextInterface = {
  castVoteRecordFiles: CastVoteRecordFiles.empty,
  electionDefinition: undefined,
  configuredAt: '',
  isOfficialResults: false,
  printBallotRef: undefined,
  saveCastVoteRecordFiles: () => undefined,
  saveElection: () => undefined,
  setCastVoteRecordFiles: () => undefined,
  saveIsOfficialResults: () => undefined,
  usbDriveStatus: UsbDriveStatus.notavailable,
  usbDriveEject: () => undefined,
  addPrintedBallot: () => undefined,
  printedBallots: [],
  fullElectionTally: getEmptyFullElectionTally(),
  fullElectionExternalTally: undefined,
  externalVoteRecordsFile: undefined,
  setFullElectionTally: () => undefined,
  setFullElectionExternalTally: () => undefined,
  saveExternalVoteRecordsFile: () => undefined,
  isTabulationRunning: false,
  setIsTabulationRunning: () => undefined,
}

const AppContext = createContext(appContext)

export default AppContext
