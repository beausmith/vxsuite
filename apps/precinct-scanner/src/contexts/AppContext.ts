import { ElectionDefinition, PrecinctId } from '@votingworks/types';
import { createContext } from 'react';
import { MachineConfig } from '../config/types';

export interface AppContextInterface {
  electionDefinition?: ElectionDefinition;
  machineConfig: Readonly<MachineConfig>;
  currentPrecinctId?: PrecinctId;
}

const appContext: AppContextInterface = {
  electionDefinition: undefined,
  currentPrecinctId: undefined,
  machineConfig: {
    machineId: '0000',
    codeVersion: '',
    bypassAuthentication: false,
  },
};

export const AppContext = createContext(appContext);
