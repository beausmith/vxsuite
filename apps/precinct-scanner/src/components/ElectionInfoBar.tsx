import React, { useContext } from 'react'

import { Prose, Text, contrastTheme, NoWrap } from '@votingworks/ui'
import { getPrecinctById } from '@votingworks/types'
import { Bar, BarSpacer } from './Bar'
import { localeWeekdayAndDate } from '../utils/IntlDateTimeFormats'
import AppContext from '../contexts/AppContext'

export type InfoBarMode = 'voter' | 'pollworker' | 'admin'

interface Props {
  mode?: InfoBarMode
}
const ElectionInfoBar: React.FC<Props> = ({ mode = 'voter' }) => {
  const { electionDefinition, currentPrecinctId, machineConfig } = useContext(
    AppContext
  )
  if (!electionDefinition) {
    return null
  }
  const electionDate = localeWeekdayAndDate.format(
    new Date(electionDefinition.election.date)
  )
  const precinct =
    currentPrecinctId !== undefined
      ? getPrecinctById({
          election: electionDefinition.election,
          precinctId: currentPrecinctId,
        })
      : undefined
  return (
    <Bar theme={{ ...contrastTheme.dark }}>
      <Prose maxWidth={false} compact>
        <NoWrap as="strong">{electionDefinition.election.title}</NoWrap> —{' '}
        <NoWrap>{electionDate}</NoWrap>
        <Text as="div" small>
          <NoWrap>{precinct?.name ?? 'All Precincts'}</NoWrap> —{' '}
          <NoWrap>
            {electionDefinition.election.county.name},{' '}
            {electionDefinition.election.state}
          </NoWrap>
        </Text>
      </Prose>
      <BarSpacer />
      {mode !== 'voter' && (
        <React.Fragment>
          <Prose maxWidth={false} compact textRight>
            <Text as="div" small>
              Software Version
            </Text>
            <strong>{machineConfig.codeVersion}</strong>
          </Prose>
          <Prose maxWidth={false} compact textRight>
            <Text as="div" small>
              Machine ID
            </Text>
            <strong>{machineConfig.machineId}</strong>
          </Prose>
        </React.Fragment>
      )}
      <Prose maxWidth={false} compact textRight>
        <Text as="div" small>
          Election ID
        </Text>
        <strong>{electionDefinition.electionHash.slice(0, 10)}</strong>
      </Prose>
    </Bar>
  )
}

export default ElectionInfoBar
