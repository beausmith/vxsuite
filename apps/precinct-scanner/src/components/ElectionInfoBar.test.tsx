import React from 'react'
import { render } from '@testing-library/react'
import { electionSampleDefinition } from '@votingworks/fixtures'
import ElectionInfoBar from './ElectionInfoBar'
import AppContext from '../contexts/AppContext'

test('Renders ElectionInfoBar', async () => {
  const { container } = render(
    <AppContext.Provider
      value={{
        electionDefinition: electionSampleDefinition,
        machineConfig: { machineId: '0000', codeVersion: 'DEV' },
      }}
    >
      <ElectionInfoBar />
    </AppContext.Provider>
  )
  expect(container).toMatchSnapshot()
})

test('Renders admin ElectionInfoBar with precinct set', async () => {
  const { container } = render(
    <AppContext.Provider
      value={{
        electionDefinition: electionSampleDefinition,
        currentPrecinctId: '23',
        machineConfig: { machineId: '0002', codeVersion: 'DEV' },
      }}
    >
      <ElectionInfoBar mode="admin" />
    </AppContext.Provider>
  )
  expect(container).toMatchSnapshot()
})
