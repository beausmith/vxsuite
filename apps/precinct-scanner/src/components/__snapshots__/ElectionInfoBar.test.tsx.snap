// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ElectionInfoBar 1`] = `
<div>
  <div
    class="sc-ksdxAp jvhYGX"
  >
    <div
      class="sc-iCfLBT bmLaUT"
    >
      <strong
        class="sc-egiSv efaHyy"
      >
        General Election
      </strong>
       —
       
      <span
        class="sc-egiSv efaHyy"
      >
        Tuesday, November 3, 2020
      </span>
      <div
        class="sc-crHlIS TTqOp"
      >
        <span
          class="sc-egiSv efaHyy"
        >
          All Precincts
        </span>
         —
         
        <span
          class="sc-egiSv efaHyy"
        >
          Franklin County
          ,
           
          State of Hamilton
        </span>
      </div>
    </div>
    <div
      class="sc-hBURRC kpEZax"
    />
    <div
      class="sc-iCfLBT bBUTlz"
    >
      <div
        class="sc-crHlIS TTqOp"
      >
        Election ID
      </div>
      <strong>
        2f6b1553c7
      </strong>
    </div>
  </div>
</div>
`;
